cluster-autoscaler/cloudprovider/aws/aws_wrapper.go

```diff
@@ -326,15 +326,15 @@ func (m *awsWrapper) getInstanceTypeFromInstanceRequirements(imageId string, req
 	}
 
 	start = time.Now()
-	instanceTypes := []string{}
+	var instanceTypes []string
 	err = m.GetInstanceTypesFromInstanceRequirementsPages(requirementsInput, func(page *ec2.GetInstanceTypesFromInstanceRequirementsOutput, isLastPage bool) bool {
 		for _, instanceType := range page.InstanceTypes {
 			instanceTypes = append(instanceTypes, *instanceType.InstanceType)
 		}
 		return !isLastPage
 	})
 	observeAWSRequest("GetInstanceTypesFromInstanceRequirements", err, start)
-	if err != nil {
+	if err != nil || len(instanceTypes) == 0 {
```

Comment:
I wonder if it would be better to have two different conditionals here: one that checks if `err != nil`, and one that checks if the number of returned instance types is 0?  As it currently stands, someone reading the logs can't tell whether things failed because of, say, a network error versus "the cloud provider couldn't find any matching instances".

Bonus points if we wrap the returned error with `%w` in the `err != nil` case so we can see exactly what went wrong on the cloud provider's end.

Reply:
make sense to me